workers/pipeline/build_nodejs.go

```diff
@@ -0,0 +1,115 @@
+package pipeline
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gaia-pipeline/gaia"
+	"github.com/gaia-pipeline/gaia/services"
+	uuid "github.com/satori/go.uuid"
+)
+
+const nodeJSInternalCloneFolder  = "jsclone"
+
+// BuildPipelineNodeJS is the real implementation of BuildPipeline for NodeJS
+type BuildPipelineNodeJS struct {
+	Type gaia.PipelineType
+}
+
+// PrepareEnvironment prepares the environment before we start the build process.
+func (b *BuildPipelineNodeJS) PrepareEnvironment(p *gaia.CreatePipeline) error {
+	// create uuid for destination folder
+	uuid := uuid.Must(uuid.NewV4(), nil)
```

Comment:
Shouldn't we generalise to use `GenerateRandomUUIDV5`?

Reply:
Good idea. We can do that after merge and then for all builders?

Reply:
Sure. Create a ticket. :D